lib/kino_vega_lite/chart_cell.ex

```diff
@@ -297,36 +299,45 @@ defmodule KinoVegaLite.ChartCell do
         name: encode(attrs.x_field),
         module: attrs.vl_alias,
         args:
-          build_arg_field(
-            attrs.x_field,
-            attrs.x_field_type,
-            attrs.x_field_aggregate,
-            attrs.x_field_bin
-          )
+          if attrs.x_field do
+            build_arg_field(
+              attrs.x_field,
+              attrs.x_field_type,
+              attrs.x_field_aggregate,
+              attrs.x_field_bin
+            )
+            |> build_scale_type_field(attrs.x_field_scale_type)
+          end
       },
       %{
         field: :y,
         name: encode(attrs.y_field),
         module: attrs.vl_alias,
         args:
-          build_arg_field(
-            attrs.y_field,
-            attrs.y_field_type,
-            attrs.y_field_aggregate,
-            attrs.y_field_bin
-          )
+          if attrs.y_field do
+            build_arg_field(
+              attrs.y_field,
+              attrs.y_field_type,
+              attrs.y_field_aggregate,
+              attrs.y_field_bin
+            )
+            |> build_scale_type_field(attrs.y_field_scale_type)
+          end
       },
       %{
         field: :color,
         name: encode(attrs.color_field),
         module: attrs.vl_alias,
         args:
-          build_arg_field(
-            attrs.color_field,
-            attrs.color_field_type,
-            attrs.color_field_aggregate,
-            attrs.color_field_bin
-          )
+          if attrs.color_field do
+            build_arg_field(
+              attrs.color_field,
+              attrs.color_field_type,
+              attrs.color_field_aggregate,
+              attrs.color_field_bin
+            )
+            |> build_scale_scheme_field(attrs.color_field_scale_scheme)
```

Comment:
Perhaps we could pass all possible options as keyword list, then filter out nils as we already do in `build_arg_field`?

```suggestion
            build_arg_field(attrs.color_field,
              type: attrs.color_field_type,
              aggregate: attrs.color_field_aggregate,
              bin: attrs.color_field_bin,
              scale: if(scheme = attrs.color_field_scale_scheme, do: [scheme: scheme])
            )
```